File: backend/src/main/java/com/example/dat251_greengafl/service/RecipeService.java
package com.example.dat251_greengafl.service;

import com.example.dat251_greengafl.the_meal_db_client.Client;
import com.example.dat251_greengafl.the_meal_db_client.Client.DetailedRecipe;
import com.example.dat251_greengafl.the_meal_db_client.Client.Recipe;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class RecipeService {

    @Autowired
    private Client client;

    public List<String> getAllCategories() {
        return client.getAllCategories();
    }

    public List<Recipe> getRecipesByCategory(String category) {
        return client.getRecipesByCategory(category);
    }

    public List<Recipe> getAllRecipes() {
        return client.getAllRecipes();
    }

    public DetailedRecipe getRecipeDetails(String id) {
        return client.getRecipeDetails(id);
    }

    public List<DetailedRecipe> searchRecipesByName(String name) {
        return client.searchRecipesByName(name);
    }

    public DetailedRecipe getRandomRecipe() {
        return client.getRandomRecipe();
    }


}

File: frontend/src/services/recipeService.ts
import { recipeApi } from "./apiClient";
import type { Recipe, DetailedRecipe } from "../api/api";

export async function getAllRecipes(): Promise<Recipe[]> {
  const response = await recipeApi.getAllRecipes();
  return response.data;
}

export async function getRecipeById(id: string): Promise<DetailedRecipe> {
  const response = await recipeApi.getRecipeById(id);
  return response.data;
}

export async function getRecipesByCategory(
  category: string
): Promise<Recipe[]> {
  const response = await recipeApi.getRecipesByCategory(category);
  return response.data;
}

export async function searchRecipesByName(
  name: string
): Promise<DetailedRecipe[]> {
  const response = await recipeApi.searchRecipesByName(name);
  return response.data;
}

export async function getRandomRecipe(): Promise<DetailedRecipe> {
  const response = await recipeApi.getRandomRecipe();
  return response.data;
}

export async function getAllCategories(): Promise<string[]> {
  const response = await recipeApi.getAllCategories();
  return response.data;
}